tests/testsuite/git_shallow.rs

```diff
@@ -0,0 +1,831 @@
+use crate::git_gc::find_index;
+use cargo_test_support::registry::Package;
+use cargo_test_support::{basic_manifest, git, paths, project};
+
+enum RepoMode {
+    Shallow,
+    Complete,
+}
+
+#[cargo_test]
```

Comment:
Not really your fault. I found git-related tests are all a bit hard to follow, because you need to understand `git2` and each helper functions beforehand. I wonder if there is a way to make it simple to learn. Maybe add some equivalent git command in comments to explain what it tries to do?

Not a blocker. I don't expect we to do more sizable updates in this PR 😆.

Reply:
> Not a blocker. I don't expect we to do more sizable updates in this PR 😆

Maybe no change is done in this PR, but I'd love to do better. My only means of describing what's going on right now is a wordy title for the `#[test]` function, and messages as last argument in assertions. Maybe a leading paragraph could describe the journey from start to end?